specs/interop/superchain-weth.md

```diff
@@ -0,0 +1,199 @@
+In its role as a native asset, ETH can be used to pay for transaction fees and can be transferred from account to
+account via calls with attached value. ETH plays a significant role in the economics of most L2s and any protocol that
+enables interoperability between chains must be able to account for ETH.
+
+### Handling native assets other than ETH
+
+Not all chains using the OP Stack use ETH as the native asset. We would like these chains to be able to interoperate
+with chains that *do* use ETH as a native asset. Certain solutions that might work when all chains use ETH as a native
+asset begin to break down when alternative native assets are introduced. For example, a protocol that burns the native
+asset on one chain and mints it on another will work if both chains use the same native asset but will obviously fail if
+either chain uses a different native asset.
+
+### Minimizing protocol complexity
+
+Support for native ETH opens the door to unnecessary complexity. Any solution to this problem should aim to minimize the
+amount of protocol code required to support native ETH. This generally points towards an app-layer solution if possible
+but does not preclude a protocol-layer solution as long as we minimize implementation size.
+
+## Constants
+
+| Name                     | Value                                        |
+| ------------------------ | -------------------------------------------- |
+| `SuperchainWETH` Address | `0x4200000000000000000000000000000000000024` |
+| `ETHLiquidity` Address   | `0x4200000000000000000000000000000000000025` |
+
+## SuperchainWETH
+
+### Invariants
+
+#### `deposit`
+
+- Reverts if triggered on a chain that does not use ETH as a native token.
+
+#### `withdraw`
+
+- Reverts if triggered on a chain that does not use ETH as a native token.
+
+#### `sendERC20`
+
+- Reverts if attempting to send more than the sender's available balance.
+- Reduce's the sender's balance by the sent amount.
+- Emits a transfer event from sender to null address for the sent amount.
+- Burns liquidity by sending the sent amount of ETH into the `ETHLiquidity` contract if native token is ETH.
+  - Must not revert.
+- Sends a message to `SuperchainWETH` on the recipient chain finalizing a send of WETH.
+  - Must not revert.
+
+#### `sendERC20To`
+
+- All invariants of `sendERC20`.
+- Message sent to `SuperchainWETH` on recipient chain includes `to` address as recipient.
+
+#### `finalizeSendERC20`
+
+- Reverts if called by any address other than the `L2ToL2CrossDomainMessenger`.
+- Reverts if `crossDomainMessageSender` is not the `SuperchainWETH` contract.
+- Mints liquidity from the `ETHLiquidity` contract if native token is ETH.
+  - Must not revert.
+- Increases the recipient's balance by the sent amount.
+- Emits a transfer event from null address to recipient for the sent amount.
+
+### Implementation Sketch
```

Comment:
I think that we can delete this implementation sketch now that we have the actual contract implemented. Some of the other specs still have source code in them for pseudo code purposes of specific functions